swap/src/bin/swap.rs

```diff
@@ -42,35 +42,11 @@ extern crate prettytable;
 async fn main() -> Result<()> {
     let args = Arguments::from_args();
 
-    let is_terminal = atty::is(atty::Stream::Stderr);
-    let base_subscriber = |level| {
-        FmtSubscriber::builder()
-            .with_writer(std::io::stderr)
-            .with_ansi(is_terminal)
-            .with_target(false)
-            .with_env_filter(format!("swap={}", level))
```

Comment:
Please use the `with_env_filter` for terminal printing in the new implementation as well! Otherwise the debug print for the terminal looks like this:

```
2021-04-06T10:49:22.293440+10:00 DEBUG Opening database at /Users/dakami/TEMP/swap/bob/01/database
2021-04-06T10:49:22.492383+10:00 DEBUG Reading in seed from /Users/dakami/TEMP/swap/bob/01/seed.pem
2021-04-06T10:49:28.264779+10:00 DEBUG Starting monero-wallet-rpc on port 50462
2021-04-06T10:49:30.040745+10:00 DEBUG connecting to 127.0.0.1:50462
2021-04-06T10:49:30.041971+10:00 DEBUG connected to 127.0.0.1:50462
2021-04-06T10:49:30.043520+10:00 DEBUG flushed 176 bytes
2021-04-06T10:49:30.044778+10:00 DEBUG parsed 5 headers
2021-04-06T10:49:30.044851+10:00 DEBUG incoming body is content-length (103 bytes)
2021-04-06T10:49:30.045013+10:00 DEBUG incoming body completed
2021-04-06T10:49:30.045431+10:00 DEBUG pooling idle connection for ("http", 127.0.0.1:50462)
2021-04-06T10:49:30.045627+10:00 DEBUG get_version RPC response: {
  "id": "1",
  "jsonrpc": "2.0",
  "result": {
    "release": true,
    "version": 65556
  }
}
2021-04-06T10:49:30.046670+10:00 DEBUG connecting to 127.0.0.1:50462
2021-04-06T10:49:30.046938+10:00 DEBUG connected to 127.0.0.1:50462
2021-04-06T10:49:30.047558+10:00 DEBUG flushed 228 bytes
2021-04-06T10:49:31.705720+10:00 DEBUG parsed 5 headers
2021-04-06T10:49:31.705808+10:00 DEBUG incoming body is content-length (59 bytes)
2021-04-06T10:49:31.705952+10:00 DEBUG incoming body completed
2021-04-06T10:49:31.706300+10:00 DEBUG pooling idle connection for ("http", 127.0.0.1:50462)
2021-04-06T10:49:31.706682+10:00 DEBUG open wallet RPC response: {
  "id": "1",
  "jsonrpc": "2.0",
  "result": {
  }
}
2021-04-06T10:49:31.706753+10:00 DEBUG Opened Monero wallet swap-tool-blockchain-monitoring-wallet
2021-04-06T10:49:31.706969+10:00 DEBUG reuse idle connection for ("http", 127.0.0.1:50462)
2021-04-06T10:49:31.709494+10:00 DEBUG flushed 193 bytes
2021-04-06T10:49:42.329512+10:00 DEBUG parsed 5 headers
2021-04-06T10:49:42.329637+10:00 DEBUG incoming body is content-length (403 bytes)
2021-04-06T10:49:42.329935+10:00 DEBUG incoming body completed
2021-04-06T10:49:42.330304+10:00 DEBUG pooling idle connection for ("http", 127.0.0.1:50462)
2021-04-06T10:49:42.330497+10:00 DEBUG get address RPC response: {
  "id": "1",
  "jsonrpc": "2.0",
  "result": {
    "address": "5BD7gmPTj6GchKqszYnAHTH3ZATujYLE63WDqoFQeLAw63CAhRG2U2QRq4sBhLjR4CfTnRDtxk7RFEeLia8VGMFNPWkuX6c",
    "addresses": [{
      "address": "5BD7gmPTj6GchKqszYnAHTH3ZATujYLE63WDqoFQeLAw63CAhRG2U2QRq4sBhLjR4CfTnRDtxk7RFEeLia8VGMFNPWkuX6c",
      "address_index": 0,
      "label": "Primary account",
      "used": false
    }]
  }
}
2021-04-06T10:49:42.335824+10:00 DEBUG Requesting quote
```

Reply:
One more note: The logfile is actually not restriceted to `debug`, but contains `TRACE` level logs as well (Not exactly sure why at this point):

```
[2mApr 06 10:49:22.285[0m [34mDEBUG[0m Opening database at /Users/dakami/TEMP/swap/bob/01/database
[2mApr 06 10:49:22.492[0m [34mDEBUG[0m Reading in seed from /Users/dakami/TEMP/swap/bob/01/seed.pem
[2mApr 06 10:49:28.264[0m [34mDEBUG[0m Starting monero-wallet-rpc on port 50462
[2mApr 06 10:49:30.040[0m [35mTRACE[0m checkout waiting for idle connection: ("http", 127.0.0.1:50462)
[2mApr 06 10:49:30.040[0m [35mTRACE[0m Http::connect; scheme=Some("http"), host=Some("127.0.0.1"), port=Some(Port(50462))
[2mApr 06 10:49:30.040[0m [34mDEBUG[0m connecting to 127.0.0.1:50462
[2mApr 06 10:49:30.041[0m [34mDEBUG[0m connected to 127.0.0.1:50462
[2mApr 06 10:49:30.042[0m [35mTRACE[0m client handshake Http1
[2mApr 06 10:49:30.042[0m [35mTRACE[0m handshake complete, spawning background dispatcher task
[2mApr 06 10:49:30.042[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Busy }
[2mApr 06 10:49:30.042[0m [35mTRACE[0m checkout dropped for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:30.043[0m [35mTRACE[0m [1mencode_headers[0m: Client::encode method=POST, body=Some(Known(61))
[2mApr 06 10:49:30.043[0m [35mTRACE[0m sized write, len = 61
[2mApr 06 10:49:30.043[0m [34mDEBUG[0m flushed 176 bytes
[2mApr 06 10:49:30.043[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: KeepAlive, keep_alive: Busy }
[2mApr 06 10:49:30.044[0m [35mTRACE[0m Conn::read_head
[2mApr 06 10:49:30.044[0m [35mTRACE[0m [1mparse_headers[0m: Response.parse([Header; 100], [u8; 263])
[2mApr 06 10:49:30.044[0m [35mTRACE[0m [1mparse_headers[0m: Response.parse Complete(160)
[2mApr 06 10:49:30.044[0m [34mDEBUG[0m parsed 5 headers
[2mApr 06 10:49:30.044[0m [34mDEBUG[0m incoming body is content-length (103 bytes)
[2mApr 06 10:49:30.044[0m [35mTRACE[0m decode; state=Length(103)
[2mApr 06 10:49:30.044[0m [34mDEBUG[0m incoming body completed
[2mApr 06 10:49:30.045[0m [35mTRACE[0m maybe_notify; read_from_io blocked
[2mApr 06 10:49:30.045[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:30.045[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:30.045[0m [35mTRACE[0m put; add idle connection for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:30.045[0m [34mDEBUG[0m pooling idle connection for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:30.045[0m [34mDEBUG[0m get_version RPC response: {
  "id": "1",
  "jsonrpc": "2.0",
  "result": {
    "release": true,
    "version": 65556
  }
}
[2mApr 06 10:49:30.045[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:30.045[0m [35mTRACE[0m pool closed, canceling idle interval
[2mApr 06 10:49:30.045[0m [35mTRACE[0m client tx closed
[2mApr 06 10:49:30.045[0m [35mTRACE[0m State::close_read()
[2mApr 06 10:49:30.046[0m [35mTRACE[0m State::close_write()
[2mApr 06 10:49:30.046[0m [35mTRACE[0m flushed({role=client}): State { reading: Closed, writing: Closed, keep_alive: Disabled }
[2mApr 06 10:49:30.046[0m [35mTRACE[0m shut down IO complete
[2mApr 06 10:49:30.046[0m [35mTRACE[0m checkout waiting for idle connection: ("http", 127.0.0.1:50462)
[2mApr 06 10:49:30.046[0m [35mTRACE[0m Http::connect; scheme=Some("http"), host=Some("127.0.0.1"), port=Some(Port(50462))
[2mApr 06 10:49:30.046[0m [34mDEBUG[0m connecting to 127.0.0.1:50462
[2mApr 06 10:49:30.046[0m [34mDEBUG[0m connected to 127.0.0.1:50462
[2mApr 06 10:49:30.047[0m [35mTRACE[0m client handshake Http1
[2mApr 06 10:49:30.047[0m [35mTRACE[0m handshake complete, spawning background dispatcher task
[2mApr 06 10:49:30.047[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Busy }
[2mApr 06 10:49:30.047[0m [35mTRACE[0m checkout dropped for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:30.047[0m [35mTRACE[0m [1mencode_headers[0m: Client::encode method=POST, body=Some(Known(112))
[2mApr 06 10:49:30.047[0m [35mTRACE[0m sized write, len = 112
[2mApr 06 10:49:30.047[0m [34mDEBUG[0m flushed 228 bytes
[2mApr 06 10:49:30.047[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: KeepAlive, keep_alive: Busy }
[2mApr 06 10:49:31.705[0m [35mTRACE[0m Conn::read_head
[2mApr 06 10:49:31.705[0m [35mTRACE[0m [1mparse_headers[0m: Response.parse([Header; 100], [u8; 218])
[2mApr 06 10:49:31.705[0m [35mTRACE[0m [1mparse_headers[0m: Response.parse Complete(159)
[2mApr 06 10:49:31.705[0m [34mDEBUG[0m parsed 5 headers
[2mApr 06 10:49:31.705[0m [34mDEBUG[0m incoming body is content-length (59 bytes)
[2mApr 06 10:49:31.705[0m [35mTRACE[0m decode; state=Length(59)
[2mApr 06 10:49:31.705[0m [34mDEBUG[0m incoming body completed
[2mApr 06 10:49:31.706[0m [35mTRACE[0m maybe_notify; read_from_io blocked
[2mApr 06 10:49:31.706[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:31.706[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:31.706[0m [35mTRACE[0m put; add idle connection for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:31.706[0m [34mDEBUG[0m pooling idle connection for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:31.706[0m [34mDEBUG[0m open wallet RPC response: {
  "id": "1",
  "jsonrpc": "2.0",
  "result": {
  }
}
[2mApr 06 10:49:31.706[0m [34mDEBUG[0m Opened Monero wallet swap-tool-blockchain-monitoring-wallet
[2mApr 06 10:49:31.706[0m [35mTRACE[0m take? ("http", 127.0.0.1:50462): expiration = Some(90s)
[2mApr 06 10:49:31.706[0m [34mDEBUG[0m reuse idle connection for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:31.709[0m [35mTRACE[0m [1mencode_headers[0m: Client::encode method=POST, body=Some(Known(78))
[2mApr 06 10:49:31.709[0m [35mTRACE[0m sized write, len = 78
[2mApr 06 10:49:31.709[0m [34mDEBUG[0m flushed 193 bytes
[2mApr 06 10:49:31.709[0m [35mTRACE[0m idle interval checking for expired
[2mApr 06 10:49:31.709[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: KeepAlive, keep_alive: Busy }
[2mApr 06 10:49:42.329[0m [35mTRACE[0m Conn::read_head
[2mApr 06 10:49:42.329[0m [35mTRACE[0m [1mparse_headers[0m: Response.parse([Header; 100], [u8; 563])
[2mApr 06 10:49:42.329[0m [35mTRACE[0m [1mparse_headers[0m: Response.parse Complete(160)
[2mApr 06 10:49:42.329[0m [34mDEBUG[0m parsed 5 headers
[2mApr 06 10:49:42.329[0m [34mDEBUG[0m incoming body is content-length (403 bytes)
[2mApr 06 10:49:42.329[0m [35mTRACE[0m decode; state=Length(403)
[2mApr 06 10:49:42.329[0m [34mDEBUG[0m incoming body completed
[2mApr 06 10:49:42.330[0m [35mTRACE[0m maybe_notify; read_from_io blocked
[2mApr 06 10:49:42.330[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:42.330[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:42.330[0m [35mTRACE[0m put; add idle connection for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:42.330[0m [34mDEBUG[0m pooling idle connection for ("http", 127.0.0.1:50462)
[2mApr 06 10:49:42.330[0m [34mDEBUG[0m get address RPC response: {
  "id": "1",
  "jsonrpc": "2.0",
  "result": {
    "address": "5BD7gmPTj6GchKqszYnAHTH3ZATujYLE63WDqoFQeLAw63CAhRG2U2QRq4sBhLjR4CfTnRDtxk7RFEeLia8VGMFNPWkuX6c",
    "addresses": [{
      "address": "5BD7gmPTj6GchKqszYnAHTH3ZATujYLE63WDqoFQeLAw63CAhRG2U2QRq4sBhLjR4CfTnRDtxk7RFEeLia8VGMFNPWkuX6c",
      "address_index": 0,
      "label": "Primary account",
      "used": false
    }]
  }
}
[2mApr 06 10:49:42.330[0m [35mTRACE[0m flushed({role=client}): State { reading: Init, writing: Init, keep_alive: Idle }
[2mApr 06 10:49:42.335[0m [34mDEBUG[0m Requesting quote
[2mApr 06 10:49:46.376[0m [33m WARN[0m Failed to dial Alice at /ip4/192.168.1.7/tcp/8765/p2p/12D3KooWPZ69DRp4wbGB3wJsxxsg1XW1EVZ2evtVwcARCF3a1nrx: Pending connection: Transport error: select error: Protocol error: I/O error: Broken pipe (os error 32)
[2mApr 06 10:49:46.377[0m [33m WARN[0m Communication error: protocol with peer 12D3KooWPZ69DRp4wbGB3wJsxxsg1XW1EVZ2evtVwcARCF3a1nrx failed due to DialFailure
[2mApr 06 10:49:46.384[0m [35mTRACE[0m client tx closed
[2mApr 06 10:49:46.385[0m [35mTRACE[0m State::close_read()
[2mApr 06 10:49:46.385[0m [35mTRACE[0m State::close_write()
[2mApr 06 10:49:46.385[0m [35mTRACE[0m pool closed, canceling idle interval
[2mApr 06 10:49:46.385[0m [35mTRACE[0m flushed({role=client}): State { reading: Closed, writing: Closed, keep_alive: Disabled }
```

Reply:
oh dear this doesn't look good

Reply:
Need to make sure to disable ANSI for the logfile!

Reply:
fixed!